scripts/broad_dsde_workflows/VariantRecalibrationWf_170305.wdl

```diff
@@ -0,0 +1,217 @@
+	    File tranches_file = "${output_basename}.${mode}.tranches"
+	    File rscript_file = "${output_basename}.${mode}.plots.R"
+    }
+}
+
+# Apply recalibration
+task ApplyRecalibrationFilter {
+    File ref_dict
+    File ref_fasta 
+    File ref_fasta_index
+    File cohort_vcf
+    File cohort_vcf_index
+    File recal_file
+    File recal_file_index
+    File interval_list
+    String output_basename
+    String mode
+    File tranches_file
+    Float filter_level
+    Int disk_size
+    String mem_size
+
+    command {
+		java -XX:GCTimeLimit=50 -XX:GCHeapFreeLimit=10 -Xmx8000m \
+        	-jar /usr/gitc/GATK36.jar \
+			-T ApplyRecalibration \
+			-R ${ref_fasta} \
+			-input ${cohort_vcf} \
+            -L ${interval_list} \
+	        -mode ${mode} \
+	        --ts_filter_level ${filter_level} \
+	        -recalFile ${recal_file} \
+	        -tranchesFile ${tranches_file} \
+	        -o ${output_basename}.vcf.gz
+    }
+
+	runtime {
+	    docker: "broadinstitute/genomes-in-the-cloud:2.2.4-1469632282"
+	    memory: mem_size
+	    disks: "local-disk " + disk_size + " HDD"
+	}
+
+    output {
+	    File recalibrated_vcf = "${output_basename}.vcf.gz"
+	    File recalibrated_vcf_index = "${output_basename}.vcf.gz.tbi"
+    }
+}
+
+workflow VariantRecalibrationWf {
+  	File ref_fasta
+  	File ref_fasta_index
+  	File ref_dict
+	File input_vcf
+	File input_vcf_index
+    File calling_intervals_list
+	Array[String] SNP_annotations
+	Array[String] INDEL_annotations
+	Array[Float] SNP_tranches
+	Array[Float] INDEL_tranches
+	Array[String] SNP_resources
+	Array[String] INDEL_resources
+	Array[File] resource_files
+	Array[File] resource_indices
+    String cohort_vcf_name
+    Float SNP_filter_level
+    Float INDEL_filter_level
+
+    # Build SNP model 
+    call BuildVQSRModel as BuildVQSRModelForSNPs {
+	    input:
+		  	ref_dict = ref_dict,
+			ref_fasta = ref_fasta,
+			ref_fasta_index = ref_fasta_index,
+		    cohort_vcf = input_vcf,
+    		cohort_vcf_index = input_vcf_index,
+            interval_list = calling_intervals_list,
+    		output_basename = cohort_vcf_name,
+    		annotations = SNP_annotations,
+    		mode = "SNP",
+    		tranches = SNP_tranches,
+    		resources = SNP_resources,
+			resource_files = resource_files,
+			resource_indices = resource_indices
+    }
+
+    # Build INDEL model 
+    call BuildVQSRModel as BuildVQSRModelForINDELs {
+	    input:
+		  	ref_dict = ref_dict,
+			ref_fasta = ref_fasta,
+			ref_fasta_index = ref_fasta_index,
+		    cohort_vcf = input_vcf,
+    		cohort_vcf_index = input_vcf_index,
+            interval_list = calling_intervals_list,
+    		output_basename = cohort_vcf_name,
+    		annotations = INDEL_annotations,
+    		mode = "INDEL",
+    		tranches = INDEL_tranches,
+    		resources = INDEL_resources,
+			resource_files = resource_files,
+			resource_indices = resource_indices
+    }
+
+    # Apply INDEL filter (first because INDEL model is usually done sooner)
```

Comment:
The two models are built in parallel, but then the recals are applied in series. With this wiring, we can already start applying Indel recalibration to the VCF while SNP model is still being built. By the time it finishes, the indel-recalibrated file is available to serve as input to apply the SNP recal. If we did it the other way around, we would have to wait until SNP recal file is available despite indel recal file being there already, then apply SNP, then apply Indel. Longer wall clock time for complete workflow execution. Switching the two apply calls bypasses this problem.